src/metabase/lib/portable.cljc

```diff
@@ -0,0 +1,77 @@
+(ns metabase.lib.portable
+  (:require
+   [medley.core :as m]
+   [metabase.lib.metadata :as lib.metadata]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.util.match :as lib.util.match]
+   [metabase.lib.walk :as lib.walk]
+   [metabase.util.malli :as mu]
+   [metabase.util.malli.registry :as mr]))
+
+(defprotocol QueryVisitor
```

Comment:
We generally try to do something like 
```suggestion
(#?(:clj p/defprotocol+ :cljs defprotocol) QueryVisitor
```

for stuff like this, that way if you reload this namespace in the REPL it won't require you to also reload every namespace that has an implementation of `QueryVisitor`.

Reply:
https://linear.app/metabase/issue/DEV-1129/add-kondo-linter-config-for-preferring-pdefrecord-and-pdefprotocol